Live Code Example (#74)
FuelLabs/docs: src/v1.0.0/1. Introduction/3. Use Cases.md

Use Cases
===

> In this section we explore various use-cases for the Fuel Version 1.0 ORU. All the use-cases covered below are examined in more detail in the [Tools and Applications](../4.%20Tools%20and%20Applications/0.%20Fast%20Withdrawals.md) section.

Reviewer: ```suggestion
In this section we explore various use cases for the Fuel [v1](../Future%20Roadmap/Planned%20Features.md#fuelv1) optimistic rollup. All the use cases covered below are examined in more detail in the [Tools and Applications](../4.%20Tools%20and%20Applications/0.%20Fast%20Withdrawals.md) section.
```

Transfers
---

> Fuel can be used for high-performance, low-latency global real-time transfers using ERC20 Ethereum compliant tokens.

Reviewer: ```suggestion
Fuel can be used for high-performance, low-latency global transfers with any ERC-20 token or Ether.
```

**This is ideal for:**
- Global payment settlement

> - Application specific tokens

Reviewer: ```suggestion
- Application-specific tokens
```

- Virtual gaming tokens and rewards

> - Any situations where users are transferring ERC20's

Reviewer: ```suggestion
- Any situations where users are transferring ERC-20 tokens
```

Please see our interactive [🕹️ Fuel Plays Pokémon demo](https://fuelplayspokemon.com) for a live example of transfers in action.

Non-Custodial Exchange
---

> Fuel's unique **multi-signature transactions and hash-time-lock contracts** are a perfect foundation for non-custodial exchanges.

Reviewer: ```suggestion
Fuel's unique **multi-user transactions and HTLC support** are a perfect foundation for non-custodial exchanges.
```

**We support two kinds of exchange:**

> - Multi-signature Swaps

Reviewer: ```suggestion
- Multi-user on-Fuel swaps
```

> - HTLC Cross-Chain Atomic Swaps

Reviewer: ```suggestion
- Cross-chain atomic swap with HTLCs
```

> For further reading, please see [Non-Custodial Exchanges](../4.%20Tools%20and%20Applications/3.%20Non-Custodial%20Exchanges.md)

Reviewer: ```suggestion
For further reading, please see [Non-Custodial Exchanges](../4.%20Tools%20and%20Applications/3.%20Non-Custodial%20Exchanges.md).
```

> Mass-Token Dispersal

Reviewer: ```suggestion
Mass Token Dispersal
```

---

> Fuel supports *multi-input, multi-output transactions*, making it perfect for pre-signed mass token dispersal scenarios where a *single account must disperse many-to-millions of tokens* to unique accounts cost-efficiently.

Reviewer: ```suggestion
Fuel supports multi-input, multi-output transactions, making it perfect for pre-signed mass token dispersal scenarios where a single account must disperse tokens to a large number of unique recipients cost-efficiently.
```

> This is possible on Fuel due to a **merklizied dispersal pattern and Fuel's unique pre-signed transaction format**.
> 
> Also note, tokens are only dispersed when the user claims them post dispersal event.

Reviewer: ```suggestion
This is possible on Fuel with a Merkleized dispersal pattern and our deterministic transaction identifiers, allowing dispersal transactions to be pre-signed and distributed off-chain. Tokens are only actually dispersed when claimed, greatly reducing overhead costs for unclaimed tokens.
```

> For further reading on how we accomplish this for near-zero cost, please see [Token Minting](../4.%20Tools%20and%20Applications/1.%20Token%20Minting.md)

Reviewer: ```suggestion
For further reading on how we accomplish this with low overheads, please see [Token Minting](../4.%20Tools%20and%20Applications/1.%20Token%20Minting.md)
```

Fast Withdrawals
---

> Fuel allows for fast (~10 minute) withdrawals using our unique HTLC output. Liquidity providers on the intended exit chain (e.g. Ethereum or Bitcoin) can allow users on Fuel to withdraw with minimal effort and at low-cost.

Reviewer: ```suggestion
Fuel allows for fast (~10 minute) withdrawals using our unique HTLC output. Liquidity providers on the intended exit chain (e.g. Ethereum or Bitcoin) can allow users on Fuel to withdraw with minimal effort and at low cost.
```

> For further reading, please see [Fast Withdrawals](../4.%20Tools%20and%20Applications/0.%20Fast%20Withdrawals.md)

Reviewer: ```suggestion
For further reading, please see [Fast Withdrawals](../4.%20Tools%20and%20Applications/0.%20Fast%20Withdrawals.md).
```

Subscriptions
---

> Fuel's pre-signed transaction format is ideal for creating regular financial subscriptions, where a user can *sign up* to a subscription, pre-signing a set of transactions, and then allowing any service or system to disperse them on a regular bases.

Reviewer: ```suggestion
Fuel's pre-signed transaction format is ideal for creating regular financial subscriptions, where a user can *sign up* to a subscription, pre-signing a set of transactions, and then allowing any service or system to collect payment on a regular basis non-interactively.
```

> This is allows for cheap, care free, subscriptions for magazines, blogs, forums and more.

Reviewer: ```suggestion
This is allows for cheap, care-free subscriptions for magazines, blogs, forums, and more.
```

> For further reading, please see [Subscriptions](../4.%20Tools%20and%20Applications/2.%20Subscriptions.md)

Reviewer: ```suggestion
For further reading, please see [Subscriptions](../4.%20Tools%20and%20Applications/2.%20Subscriptions.md).
```

Burning
---

> Fuel has a return data output, which can be used to store arbitrary transaction data in the Fuel ORU such as: burn flags, tweets, hashes or specific application data.
> 
> The burning of a token by sending it to the null or a voided address can be contextualized around a specific event using the return data output, all within in the same transaction.

Reviewer: ```suggestion
Fuel has a "return data" transaction output, which can be used to store arbitrary data in Fuel such as: burn metadata, social media messages, timestamped hashes, or specific application data.

The burning of a token by sending it to the zero address can be associated with a specific event using the return data output, all within in the same transaction.
```
